elasticai/creator/nn/quantized_grads/fixed_point/test_quantize_to_fixed_point.py

```diff
@@ -0,0 +1,45 @@
+import torch
+
+from elasticai.creator.nn.quantized_grads.fixed_point import FixedPointConfigV2
+from elasticai.creator.nn.quantized_grads.fixed_point._quantize_to_fixed_point import round_to_fixed_point, clamp, \
+    quantize
+
+def test_round_determinstic_to_fixed_point():
+    conf = FixedPointConfigV2(total_bits=8,
+                           frac_bits=2,
+                           stochastic_rounding=False)
+    x = torch.range(-2, 2, step=0.1, dtype=torch.float32)
+
+    actual = round_to_fixed_point(x, conf.frac_bits)
+    expected = torch.Tensor([-2.0000, -2.0000, -1.7500, -1.7500, -1.5000, -1.5000, -1.5000, -1.2500,
+                       -1.2500, -1.0000, -1.0000, -1.0000, -0.7500, -0.7500, -0.5000, -0.5000,
+                       -0.5000, -0.2500, -0.2500, -0.0000,  0.0000,  0.0000,  0.2500,  0.2500,
+                       0.5000,  0.5000,  0.5000,  0.7500,  0.7500,  1.0000,  1.0000,  1.0000,
+                       1.2500,  1.2500,  1.5000,  1.5000,  1.5000,  1.7500,  1.7500,  2.0000,
+                       2.0000])
+    assert(torch.equal(actual,expected))
+
+def test_clamp_to_fixed_point():
+    conf = FixedPointConfigV2(total_bits=4,
+                              frac_bits=2,
+                              stochastic_rounding=False)
+    x = torch.range(-3, 3, step=1, dtype=torch.float32)
+
+    actual = clamp(x, conf)
+    expected = torch.Tensor([-2.0000, -2.0000, -1.0000,  0.0000,  1.0000,  1.7500,  1.7500])
+    assert(torch.equal(actual, expected))
+
+def test_quantize_determinstic_to_fixed_point():
+    conf = FixedPointConfigV2(total_bits=4,
+                              frac_bits=2,
+                              stochastic_rounding=False)
```

Comment:
The *deterministic* part of unit tests sucks a bit here, right? 
I'm currently eyeballing with https://hypothesis.readthedocs.io/en/latest/
it leverages pytest and automatically generates input data for your algorithm, it will then remember for which input data your tests failed and be able to replay these. I think for all the numeric stuff we need to do in the creator and with ML it could be really powerful.

Especially recommend this article for an introduction into their stateful testing: https://hypothesis.works/articles/how-not-to-die-hard-with-hypothesis/

Reply:
Why does it suck? 
The deterministic part is that I only test deterministic rounding which implements Round half to even. 

The tool you mention sounds cool. I guess it would make sense to have look at it with a small demo. 
But should be a different PR

Reply:
> But should be a different PR

Of course it should.

I always find testing stochastic processes difficult, you want to make sure that your sample size is big enough to cover a meaningful number of cases, on the other hand you want deterministic behaviour, so you can rely on your tests.